e2e/cypress/integration/archived_channel/archive_channel_search_spec.js

```diff
@@ -0,0 +1,163 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @channel
+
+import {getAdminAccount} from '../../support/env';
+import {getRandomId} from '../../utils';
+
+describe('archive tests while preventing viewing archived channels', () => {
+    let testTeam;
+    let testChannel;
+    let testUser;
+    let adminUser;
+    const testArchivedMessage = `this is an archived post ${getRandomId()}`;
+
+    before(() => {
+        adminUser = getAdminAccount();
+
+        cy.apiUpdateConfig({
+            TeamSettings: {
+                ExperimentalViewArchivedChannels: true,
+            },
+        });
+
+        // # Login as test user and visit town-square
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            testTeam = team;
+            testChannel = channel;
+            testUser = user;
+
+            cy.apiCreateUser({prefix: 'second'}).then(({user: second}) => {
+                cy.apiAddUserToTeam(testTeam.id, second.id);
+            });
+            cy.visit(`/${team.name}/channels/${testChannel.name}`);
+            cy.postMessageAs({sender: testUser, message: testArchivedMessage, channelId: testChannel.id});
+        });
+    });
+
+    it('MM-T1707 Unarchived channels can be searched the same as before they where archived', () => {
+        cy.apiLogin(adminUser);
```

Comment:
Actually, you can safely remove this line since up until here, the session is still with the sysadmin.

fyi, there's `cy.apiAdminLogin()` that you may use conveniently.